packages/flutter/lib/src/widgets/sensitive_content.dart

```diff
@@ -0,0 +1,277 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart' show ContentSensitivity, SensitiveContentService;
+
+import '../../widgets.dart' show ConnectionState;
+import 'async.dart' show AsyncSnapshot, FutureBuilder;
+import 'container.dart';
+import 'framework.dart';
+
+/// Data structure used to track the [SensitiveContent] widgets in the
+/// widget tree.
+class ContentSensitivitySetting {
+  /// Creates a [ContentSensitivitySetting].
+  ContentSensitivitySetting();
+
+  /// The number of [SensitiveContent] widgets that have sensitivity level [ContentSensitivity.sensitive].
+  int _sensitiveWidgetCount = 0;
+
+  /// The number of [SensitiveContent] widgets that have sensitivity level [ContentSensitivity.autoSensitive].
+  int _autoSensitiveWidgetCount = 0;
+
+  /// The number of [SensitiveContent] widgets that have sensitivity level [ContentSensitivity.notSensitive].
+  int _notSensitiveWigetCount = 0;
+
+  /// Increases the count of [SensitiveContent] widgets with [sensitivityLevel] set.
+  void addWidgetWithContentSensitivity(ContentSensitivity sensitivityLevel) {
+    switch (sensitivityLevel) {
+      case ContentSensitivity.sensitive:
+        _sensitiveWidgetCount++;
+      case ContentSensitivity.autoSensitive:
+        _autoSensitiveWidgetCount++;
+      case ContentSensitivity.notSensitive:
+        _notSensitiveWigetCount++;
+    }
+  }
+
+  /// Decreases the count of [SensitiveContent] widgets with [sensitivityLevel] set.
+  void removeWidgetWithContentSensitivity(ContentSensitivity sensitivityLevel) {
+    switch (sensitivityLevel) {
+      case ContentSensitivity.sensitive:
+        _sensitiveWidgetCount--;
+      case ContentSensitivity.autoSensitive:
+        _autoSensitiveWidgetCount--;
+      case ContentSensitivity.notSensitive:
+        _notSensitiveWigetCount--;
+    }
+  }
+
+  /// Returns true if this class is currently tracking at least one [SensitiveContent] widget.
+  bool get hasWidgets =>
+      _sensitiveWidgetCount + _autoSensitiveWidgetCount + _notSensitiveWigetCount > 0;
```

Comment:
you might consider 
```suggestion
      max(0, _sensitiveWidgetCount) + max(0,_autoSensitiveWidgetCount) + max(0, _notSensitiveWigetCount) > 0;
```
to protect you in case of a bug where  you remove more widgets or you remove with the wrong level of sensitivity.